### PR TITLE
added task to capitalize recipe names with special characters

### PR DESCRIPTION
**Pull request by dshcheung**

### CODE REVIEW

**Comment by h8rry** [comment]
@dshcheung Can you give me an example of `recipe name after special characters`? Thanks!

**Comment by dshcheung** [comment]
"egg Not" (eggless Eggnog) to
"Egg Not" (Eggless Eggnog)

File: lib/tasks/capitalize-recipe-name.rake [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      index = name.index(/\(/, start_index) + 1
+      if name[index] != " " && name[index] != nil
+        name[index] = name[index].upcase
+      end
+      start_index = index
+    end
+
+    Recipe.find(recipe.id).update(name: name)
+  end
+end
```
Comment (h8rry) [inline lib/tasks/capitalize-recipe-name.rake]: Add a new line to the end of the file to avoid this

File: lib/tasks/capitalize-recipe-name.rake [hunk 2]
```diff
@@ -0,0 +1,34 @@
+desc "capitalize recipe name after special characters"
+task :capitalize_name => :environment do
+  Recipe.where("name LIKE ? or name LIKE ? or name LIKE ?", '%-%', '%(%', '%"%').find_each(batch_size: 100) do |recipe|
+    name = recipe.name
+    puts name
+
+    start_index = 0
+    while name.index(/-/, start_index) != nil
+      index = name.index(/-/, start_index) + 1
```
Comment (harrychenca) [inline lib/tasks/capitalize-recipe-name.rake]: make this a function

```ruby
 index = name.index(/-/, start_index) + 1
      if name[index] != " " && name[index] != nil
        name[index] = name[index].upcase
      end
      start_index = index
```

File: lib/tasks/capitalize-recipe-name.rake [hunk 3]
```diff
@@ -0,0 +1,34 @@
+desc "capitalize recipe name after special characters"
+task :capitalize_name => :environment do
+  Recipe.where("name LIKE ? or name LIKE ? or name LIKE ?", '%-%', '%(%', '%"%').find_each(batch_size: 100) do |recipe|
+    name = recipe.name
+    puts name
+
+    start_index = 0
```
Comment (h8rry) [inline lib/tasks/capitalize-recipe-name.rake]: make this whole thing as a function

File: lib/tasks/capitalize-recipe-name.rake [hunk 4]
```diff
@@ -0,0 +1,22 @@
+desc "capitalize recipe name after special characters"
+task :capitalize_name => :environment do
+  Recipe.where("name LIKE ? or name LIKE ? or name LIKE ?", '%-%', '%(%', '%"%').find_each(batch_size: 100) do |recipe|
+    name = recipe.name
+    puts name
+
+    special_characters = ['-', '\(', '"']
+    special_characters.each do |character|
+      start_index = 0
```
Comment (h8rry) [inline lib/tasks/capitalize-recipe-name.rake]: make this a function

File: lib/tasks/capitalize-recipe-name.rake [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    special_characters = ['-', '\(', '"']
+    name = capitalize_str(special_characters, recipe.name)
+    puts name
+    
+    Recipe.find(recipe.id).update(name: name)
+  end
+end
+
+def capitalize_str(special_characters, name)
+  special_characters.each do |character|
```
Comment (h8rry) [inline lib/tasks/capitalize-recipe-name.rake]: I think the loop shouldn't be part of the function. A function should just do a single task and be re-usable.

**Pull request merged**